# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaDevices.cpp
# Commit: 88ca4bb1602f
# Full Hash: 88ca4bb1602fae1c9dd45cf64e742618c7ede85e
# Author: Ben Lienhart <blienhar@gmu.edu>
# Date: 2022-03-16 21:49:37
# Regressor Bug: 1758367
# File Overlap Count: 1
# Description:
#   Bug 1758367 - Implemented finer-grained RFP check in MediaDevices for cut-over of ShouldRFP. r=tjr
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D140496
# ==============================================================================

diff -r 0e92d5991ca1 -r 88ca4bb1602f dom/media/MediaDevices.cpp
--- a/dom/media/MediaDevices.cpp	Wed Mar 16 19:38:00 2022 +0200
+++ b/dom/media/MediaDevices.cpp	Wed Mar 16 17:34:11 2022 +0000
@@ -689,8 +689,16 @@
 
   // Do not fire event to content script when
   // privacy.resistFingerprinting is true.
-  if (nsContentUtils::ShouldResistFingerprinting()) {
-    return;
+
+  if (nsContentUtils::ShouldResistFingerprinting(
+          "Guarding the more expensive RFP check with a simple one")) {
+    nsCOMPtr<nsIGlobalObject> global = xpc::NativeGlobal(GetWrapper());
+    nsCOMPtr<nsPIDOMWindowInner> owner = do_QueryInterface(global);
+
+    Document* doc = owner->GetExtantDoc();
+    if (nsContentUtils::ShouldResistFingerprinting(doc)) {
+      return;
+    }
   }
 
   mHaveUnprocessedDeviceListChange = true;
